build/patches/Add-lifetime-options-permissions.patch

```diff
@@ -0,0 +1,1415 @@
+@@ -153,7 +154,7 @@ void GetOrigins(JNIEnv* env,
+     seen_origins.push_back(origin);
+     insertionFunc(env, static_cast<int>(content_type), list,
+                   ConvertOriginToJavaString(env, origin), jembedder,
+-                  /*is_embargoed=*/false);
++                  /*is_embargoed=*/false, static_cast<int>(settings_it.session_model));
+   }
+ 
+   // Add any origins which have a default content setting value (thus skipped
+@@ -176,7 +177,7 @@ void GetOrigins(JNIEnv* env,
+       seen_origins.push_back(origin);
+       insertionFunc(env, static_cast<int>(content_type), list,
+                     ConvertOriginToJavaString(env, origin), jembedder,
+-                    /*is_embargoed=*/true);
++                    /*is_embargoed=*/true, 0);
+     }
+   }
+ }
+diff --git a/components/browser_ui/strings/android/browser_ui_strings.grd b/components/browser_ui/strings/android/browser_ui_strings.grd
+--- a/components/browser_ui/strings/android/browser_ui_strings.grd
++++ b/components/browser_ui/strings/android/browser_ui_strings.grd
+@@ -581,6 +581,11 @@
+       <message name="IDS_PAGE_INFO_URL_TRUNCATED" desc="Accessibility announcement when the URL in PageInfo switches from full to truncated display">
+         URL truncated
+       </message>
++      <message name="IDS_PAGE_INFO_ANDROID_PERMISSION_SESSION_PERMISSION"
++               desc="The label used in the About Page dialog to indicate an session permission">
++        (only this session)
++      </message>
++
+       <message name="IDS_PAGE_INFO_AD_PERSONALIZATION_TITLE" desc="The title of the 'Ad personalization' section in Page Info.">
+           Ad personalization
+       </message>
+diff --git a/components/content_settings/android/BUILD.gn b/components/content_settings/android/BUILD.gn
+--- a/components/content_settings/android/BUILD.gn
++++ b/components/content_settings/android/BUILD.gn
+@@ -57,6 +57,7 @@ java_cpp_enum("content_settings_enums_javagen") {
+     "../core/common/content_settings_types.h",
+     "../core/common/cookie_controls_enforcement.h",
+     "../core/common/cookie_controls_status.h",
++    "../core/common/content_settings_constraints.h",
+   ]
+   visibility = [ ":*" ]  # Depend on through :content_settings_enums_java
+ }
+diff --git a/components/content_settings/core/browser/content_settings_utils.cc b/components/content_settings/core/browser/content_settings_utils.cc
+--- a/components/content_settings/core/browser/content_settings_utils.cc
++++ b/components/content_settings/core/browser/content_settings_utils.cc
+@@ -187,4 +187,32 @@ base::Time GetConstraintExpiration(const base::TimeDelta duration) {
+   return base::Time::Now() + duration;
+ }
+ 
++ContentSettingConstraints GetConstraintSessionExpiration(LifetimeMode lifetime_mode) {
++  if (lifetime_mode == LifetimeMode::OnlyThisTime) {
++    // note: this content settings will be discarded immediately
++    return {base::Time() + base::Hours(1), content_settings::SessionModel::UserSession};
```

Comment:
If it is discarded immediately then why the +1 hour here?

Reply:
Please add a comment here and where it is evaluated about this `1h` being a magic value used to differentiate it from the others.

Reply:
in my opinion it is not needed, because the logic is only present in that file, that is, it is an opaque value for everyone

Reply:
It is necessary because it is a magic number and it is not made as a separate constant.

There should be a comment here:
```
} else if (source.expiration == (base::Time() + base::Hours(1))) {
    // 1h is used as a magic constant to identify the one-time lifetime mode
```
And also here:
```
// note: this content settings will be discarded immediately
// the 1h value is used as a magic constant here
```

Without such comments reader can think it is valid for 1h.

Reply:
fixed in https://github.com/bromite/bromite/pull/1958/commits/334744dfe5cfc2ebcfb78243978e0af531f85f31